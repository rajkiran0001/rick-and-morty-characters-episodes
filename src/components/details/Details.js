import React, {useState, useEffect} from 'react';
import axios from 'axios';

function Details(route) {
    const [data, setData] = useState({});
    const [loading, setLoading] = useState(true);

    const getCharacter = async () => {
        console.log('searching...');
        console.log('route:', route);
        const result = await axios(
            `https://rickandmortyapi.com/api/character/${route.id}`
        );
        console.log('result:', result);
        setData(result.data);
        setLoading(false);
    };

    useEffect(() => {
        getCharacter();
    }, []);

    return loading ? (
        <p>Loading...</p>

    ) : (
        <div >
            <img alt={data.image} src={data.image} />
            <h1 className="heading">Name: {data.name}</h1>
            <div >
                Gender: {data.gender} <span>Origin: ({data.origin.name})</span>
            </div>
            <div >
                <h1 className="episode">Episodes</h1>
            {data.episode.map(item => (
                <p key={item}>{item}</p>
            ))}
            </div>
        </div>
    );
}

export default Details;
