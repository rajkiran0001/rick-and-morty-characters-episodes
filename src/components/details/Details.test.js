import React from "react";
import { shallow, mount } from "enzyme";
import Details from "./Details";
import Spinner from '../../Spinner';

describe("Details Page Snapshot ", () => {
  let mountWrapper;
  beforeEach(() => {
      mountWrapper = mount(<Details />);
  })

  afterEach(() => {
      mountWrapper.unmount();
  })

  it("renders Correctly", () => {
      expect(mountWrapper).toMatchSnapshot();
  })
});

describe("Details Page rendering of element", () => {
  let wrapper;
  beforeEach(() => {
      wrapper = shallow(<Details />);
  })

  afterEach(() => {
      wrapper.unmount();
  })
  
  it("Renders one Spinner React Compount", ()=> {
      expect(wrapper.find(Spinner).render())
  })
});