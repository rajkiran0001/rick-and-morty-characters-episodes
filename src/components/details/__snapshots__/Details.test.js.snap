// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Page Snapshot  renders Correctly 1`] = `ReactWrapper {}`;
